pinot-common/src/test/java/org/apache/pinot/sql/parsers/CalciteSqlCompilerTest.java

```diff
@@ -2681,6 +2681,63 @@ public void testParserExtensionImpl() {
     Assert.assertEquals(sqlNodeAndOptions.getSqlType(), PinotSqlType.DML);
   }
 
+
+  @Test
+  public void shouldParseBasicAtTimeZoneExtension() {
```

Comment:
can we also add a test in `QueryCompilationTest` for the v2 engine. 
- currently it should throw an exception "at_time_zone" function no match signature instead of stating `<time>` is unexpected identifier
- once we add the scalar function atTimeZone we should resolve the compilation correctly